src/v/cluster/node_status_backend.cc

```diff
@@ -41,25 +51,75 @@ node_status_backend::node_status_backend(
 
     _members_table_notification_handle
       = _members_table.local().register_members_updated_notification(
-        [this](auto ids) {
-            handle_members_updated_notification(std::move(ids));
+        [this](std::vector<model::node_id> ids) {
+            for (auto& id : ids) {
+                const auto node_meta
+                  = _members_table.local().get_node_metadata_ref(id);
+                if (!node_meta) {
+                    _pending_member_notifications.emplace_back(
+                      id, model::membership_state::removed);
```

Comment:
Is this correct? The node is in the new set (`ids`), but not in the old one, why is the state `removed`?

Reply:
here the notification contains only ids, we need to check if the node was added or removed from cluste as either fo those events can trigger notification